### PR TITLE
Chore: Update crossbrowser config and dependencies

### PR DESCRIPTION
**Pull request by Rob-Doherty**
# Description

This change updates the Crossbrowser tests, and upgrades CodeceptJS, Puppeteer and Webdriver to latest or nearly latest stable versions.

As part of this, updates were made to
- follow CodeceptJS's move to Webdriver helper
- fix saucelab.conf.js's waitForTimeout reference
- update supportedBrowsers.js to test on the latest Mac and Windows OSs
- install mocha-multi to support multiple reporters
- add `allLanguages` feature toggle to be able to run tests for all languages or just English (to speed up the crossbrowser tests without losing Welsh coverage from PR pipeline tests)
- update Jenkinsfile_nightly to set features as per AAT
 
Fixes # https://tools.hmcts.net/jira/browse/TA-204

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: test/e2e/paths/respondent/2yrSeparationHappyPath-en-cy.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const config = require('config');
 const twoPlusYearsDivorceSession = require('test/resources/2PlusYears-divorce-session');
 
 Feature('2 years Separation Journey');
-const languages = ['en', 'cy'];
+const languages = (config.tests.e2e.allLanguages === 'true') ? ['en', 'cy'] : ['en'];
```
Comment (mounikahmcts) [inline test/e2e/paths/respondent/2yrSeparationHappyPath-en-cy.js:6]: file name looks odd. As we are covering it based on the condition. Can we change the file name without any languages in it.

File: config/custom-environment-variables.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ tests:
     proxy: E2E_IDAM_PROXY
     proxyByPass: E2E_IDAM_SERVER
     addWaitForCrossBrowser: E2E_ADD_WAIT_CROSSBROWSER
+    allLanguages: E2E_ALL_LANGUAGES
```
Comment (mounikahmcts) [inline config/custom-environment-variables.yaml:20]: 👍

**Review by mounikahmcts — APPROVED** [review]
looks good to me 👍

File: test/e2e/paths/respondent/2yrSeparationHappyPath-en-cy.js (hunk 1 above)
Comment (Rob-Doherty) [reply test/e2e/paths/respondent/2yrSeparationHappyPath-en-cy.js:6]: good question @mounikahmcts - my original expectation had been that the crossbrowser tests would only need to run for one language, but Satya pointed out that the functionality could be impacted and so they need to run on both. I'll remove this now

Comment (Rob-Doherty) [reply test/e2e/paths/respondent/2yrSeparationHappyPath-en-cy.js]: reverted.

**Review by satyachundur — APPROVED** [review]

**Pull request merged**